1

valzebr opened the pull request:
?

valzebr committed:
First commit2

valzebr committed:
First commit3

valzebr committed:
Почти доделал

valzebr committed:
Завтра допишу

AlexKitIt commented on data.py:
```diff
@@ -0,0 +1,51 @@
+    ]
+
+
+# Для корректного отображения аргументов в параметризированном тесте
+def pytest_make_parametrize_id(val):
+    return repr(val)
```
Нужно исправить: этот метод размещается в conftest

valzebr replied:
Исправил

AlexKitIt commented on tests/test_go_to_dzen.py:
```diff
@@ -0,0 +1,11 @@
+
+class TestDzen:
+
+    pass
```
Нужно исправить: тут должен быть тест на редирект

valzebr replied:
Этот файл поменял весь

valzebr committed:
Доделал:
Перенёс функцию в нужный файл, проверка на кнопки самоката и яндекса в отдельном файле, добавил отчёты

AlexKitIt commented on tests/tests_header_button.py:
```diff
@@ -0,0 +1,28 @@
+        correct_url = Urls.DZEN_URL
+        home_page.yandex_button_click()
+        home_page.go_to_new_tab()
+        time.sleep(2)
```
Нужно исправить: слипов и неявных ожиданий быть не должно. Необходимо использовать expected_conditions
Исправил

AlexKitIt commented on tests/tests_header_button.py:
```diff
@@ -0,0 +1,28 @@
+        home_page.yandex_button_click()
+        home_page.go_to_new_tab()
+        time.sleep(2)
+        assert driver.current_url == correct_url
```
Нужно исправить: здесь и далее: прямого обращения к driver из тестов быть не должно, только через методы пейджей
Исправил

AlexKitIt commented on tests/tests_header_button.py:
```diff
@@ -0,0 +1,28 @@
+        correct_url = Urls.BASE_URL
+        home_page.first_order_button_click()
+        home_page.scooter_logo_click()
+        assert correct_url == Urls.BASE_URL
```
Нужно исправить: в 16 строке этой переменной присваивается такое значение, в чем проверка? 
Ответ: Планировалось, что открывается вкладка и её адрес соответсвует нужному
Я так понимаю, этот подход был вкорне неверным. Если так, почему?
Сам метод исправил

valzebr committed:
Надеюсь, что это были
последние изменения :)
